src/proxyfoldermodel.cpp

```diff
@@ -156,14 +156,34 @@ bool ProxyFolderModel::lessThan(const QModelIndex& left, const QModelIndex& righ
 }
 
 std::shared_ptr<const Fm::FileInfo> ProxyFolderModel::fileInfoFromIndex(const QModelIndex& index) const {
-    FolderModel* srcModel = static_cast<FolderModel*>(sourceModel());
-    if(srcModel) {
-        QModelIndex srcIndex = mapToSource(index);
-        return srcModel->fileInfoFromIndex(srcIndex);
+    if(index.isValid()) {
+        FolderModel* srcModel = static_cast<FolderModel*>(sourceModel());
+        if(srcModel) {
+            QModelIndex srcIndex = mapToSource(index);
+            return srcModel->fileInfoFromIndex(srcIndex);
+        }
     }
     return nullptr;
 }
 
+QModelIndex ProxyFolderModel::indexFromPath(const FilePath &path) const {
```

Comment:
@palinek Thank you for your attention! `indexFromPath` also checks items one by one, so I don't think it'll make any difference. That being said, I'll surely give it a try.

Our only problem in graying out cut files is about when another app cuts them -- the current functions might take a lot of CPU time in that case. For now, we don't gray out cut files in pcmanfm-qt when that happens (as a workaround).